services/order.py

```diff
@@ -0,0 +1,35 @@
+import datetime
+from typing import Optional
+
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, User, Ticket
+
+
+@transaction.atomic()
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: Optional[datetime.datetime] = None
+) -> None:
+    user = User.objects.get(username=username)
+    order = Order.objects.create(user=user)
+
+    if date:
+        order.created_at = date
+        order.save()

```

Comment:
This code executes two SQL queries. Please, upgrade this code so that it executes one SQL query.

Reply:
if i change the code for example: order = Order.objects.create(user=user, created_at=date if date else None)   it gonna execute 1 sql query but it will result in AssertionError 
AssertionError: assert datetime.datetime(2023, 3, 30, 12, 49, 23, 153446) == datetime.datetime(2020, 11, 10, 14, 40)